src/core/plugins/oas31/oas3-extensions/fn.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * @prettier
+ */
+import { objectify } from "core/utils"
+
+export const makeIsFileUploadIntended = (getSystem) => {
+  const isFileUploadIntended = (schema, mediaType = null) => {
```

Comment:
We're looking to use `isFileUploadIntendedOAS30` here and introduce new function called `schemaHasType` both in oas3 and oas31 plugins introducing spec compliant business logic.